ERCS/erc-7007.md

```diff
@@ -133,14 +182,40 @@ ERC-7007 Metadata JSON Schema for reference
+        },
+        "proof_type": {
+            "type": "string",
+            "description": "validity (zkML) or fraud (opML)"
         }
     }
 }
 ```
 
 ## Rationale
 
-TBD
+### Unique Token Identification
+
+This specification sets the `tokenId` to be the hash of its corresponding `prompt`, creating a deterministic and collision-resistant way to associate tokens with their unique content generation parameters. This design decision ensures that the same prompt (which corresponds to the same AI-generated content under the same model seed) cannot be minted more than once, thereby preventing duplication and preserving the uniqueness of each NFT within the ecosystem.
+
+### Generalization to Different Proof Types
+
+This specification accommodates two proof types: validity proofs for zkML and fraud proofs for opML. Function arguments in `mint` and `verify` are designed for generality, allowing for compatibility with both proof systems. Moreover, the specification includes an updatable extension that specifically serves the requirements of opML.
+
+### `verify` interface
+
+We specify a `verify` interface to enforce the correctness of `aigcData`. It is defined as a view function to reduce gas cost. `verify` SHOULD return true if and only if `aigcData` is finalized in both zkML and opML. In zkML, it MUST verify the ZK proof, i.e. `proof`; in opML, it MUST make sure that the challenging period is finalized, and that the `aigcData` is up-to-date, i.e. has been updated after finalization. Additionally, `proof` CAN be *empty* in opML.
+
+### `mint` interface
+
+We specify a `mint` interface to bind the prompt and `aigcData` with `tokenId`. Notably, it acts differently in zkML and opML cases. In zkML, `mint` SHOULD make sure `verify` returns `true`. While in opML, it CAN be called before finalization. The consideration here is that, limited by the proving difficulty, zkML usually targets simple model inference tasks in practice, making it possible to provide a proof within an acceptable time frame. On the other hand, opML enables large model inference tasks, with a cost of longer confirmation time to achieve the approximate same security level. Mint until opML finalization may not be the best practice considering the existing optimistic protocols.
+
+### Naming Choice on `update`
+
+We adopt "update" over "finalize" because a successful challenge happens rarely in practice. Using `update` could avoid calling it for every `tokenId` and save gas.
+
+### ML Model Publication
+
+While this standard does not describe the Machine Learning model publication stage, it is natural and recommended to publish the commitment of the Model to Ethereum separately, before any actual `mint` actions. The model commitment schema choice lies on the AIGC-NFT project issuer party. The commitment should be checked inside the implementation of the `verify` function.
```

Comment:
> The commitment should be checked inside the implementation of the `verify` function.

This is definitely a recommendation for implementations, so needs to be in the Specification section.